scripts/Jenkinsfile

```diff
@@ -0,0 +1,181 @@
+				git_remote_add("hcc_fork/compiler-rt","compiler-rt_fork", "ssh://git@github.com/${ userID}/compiler-rt.git")
+				git_remote_add("hcc_fork/compiler-rt","compiler-rt", "ssh://git@github.com/llvm-mirror/compiler-rt.git")
+				git_remote_add("hcc_fork/clang-tools-extra","clang-tools-extra_fork", "ssh://git@github.com/${ userID}/clang-tools-extra.git")
+				git_remote_add("hcc_fork/clang-tools-extra","clang-tools-extra", "ssh://git@github.com/llvm-mirror/clang-tools-extra.git")
+				git_remote_add("hcc_fork/clang","clang_fork", "ssh://git@github.com/${ userID}/hcc-clang-upgrade.git")
+				git_remote_add("hcc_fork/clang","clang", "ssh://git@github.com/RadeonOpenCompute/clang")
+			}
+		}
+		
+		if (startStage <= 3) { 
+			stage ('Stage 3:Checkout latest master ROCM-Device-Libs') {
+				sh (script: "cd hcc_fork/rocdl; git fetch --all; git checkout origin/master")
+			}
+		}
+		
+		if (startStage <= 4) { 		
+			stage ('Stage 4: Merge amd-common  commits') {
+			    
+			    if (!fileExists("merge-branch-name") ) { 
+			    	writeFile file: "merge-branch-name", text: "${ merge_name}"
+				} 
+				merge_name = readFile "merge-branch-name"
+				
+				dir("hcc_fork") {
+					git_config()
+					git_merge("lld", merge_name, "origin/amd-hcc", "origin/amd-common")
+					git_merge("compiler",merge_name,"origin/amd-hcc", "origin/amd-common")
+					git_merge("compiler-rt", merge_name, "origin/amd-hcc", "--no-ff compiler-rt/master")
+					git_merge("clang-tools-extra", merge_name, "origin/amd-hcc", "clang-tools-extra/master")
+					git_merge("clang", merge_name, "origin/clang_tot_upgrade", "--no-ff clang/amd-common")
+				}
+			}		
+		}
+		
+		if (startStage <= 5) { 
+			stage ('Stage 5: Build merged HCC') {
+				sh (script: "mkdir -p build; cd build; cmake -DCMAKE_BUILD_TYPE=Release ../hcc_fork; make -j${ nproc}")
+			}
+		}
+		
+		if (startStage <= 6) { 
+			stage ('Stage 6: Full sanity tests on merged HCC') {
+				sh (script: "cd build; make test -j${ nproc}")
+			}
+		}
+		
+		if (startStage <= 7) { 		
+			stage ('Stage 7: Quick sanity tests on merged HCC') {
+				sh (script: "/opt/rocm/bin/hcc `/opt/rocm/bin/clamp-config --build --cxxflags --ldflags` -lm hcc_fork/tests/Unit/AmpMath/amp_math_cos.cpp; ./a.out ")
+				sh (script: "/opt/rocm/bin/hcc `/opt/rocm/bin/hcc-config --build --cxxflags --ldflags` -lhc_am hcc_fork/tests/Unit/GridLaunch/glp_const.cpp; ./a.out ")
+				sh (script: "/opt/rocm/bin/hcc `/opt/rocm/bin/hcc-config --build --cxxflags --ldflags` hcc_fork/tests/Unit/HC/hc_atomic_add_global.cpp ; ./a.out ")
+			}
+		}
+		
+		if (startStage <= 8) { 
+			stage('Stage 8: Push and Create Pull Requests') {
+			
+				if (!fileExists("merge-branch-name") ) { 
+			    	writeFile file: "merge-branch-name", text: "${ merge_name}"
+				} 
+				
+				merge_name = readFile "merge-branch-name"
+				
+				dir ("hcc_fork") {
+					dir ("clang") {
+						git_push("clang_fork", merge_name)
+						//sh (script: "git push clang_fork ${merge_name}:${merge_name}")
```

Comment:
@david-salinas these lines commented out could be removed due to the introduction of `git_push()` function?

Reply:
yes.  Though I just want to make sure the git_push function work properly, then I'll just remove them.  I also need to add a git_request-pull function, as the last step.